github-git/3-fork-repo.md

```diff
@@ -0,0 +1,100 @@
+:::
+
+
+
+## Who owns a GitHub repository?
+
+When a repository is stored on **GitHub.com**, it is assigned a unique URL (i.e. link on the **GitHub.com** website) that can be used to find the repository and access its files. While repositories on **GitHub.com** can be made either public or private, the default is public for free **GitHub** accounts.
+
+In either case (public or private), the URL links to a **GitHub** repository always follows the same format: 
+
+`https://github.com/username/repository-name`
+
+The username is the username owner of the repo owner. The username can either be an individual such as `lwasser` (your **GitHub** username), or it can represent an organization such as `pyOpenSci`.
+
+For example, the repository that you will work within this lesson is owned by `pyopensci.` Therefore, the URL looks like this:
+
+`https://github.com/pyopensci/repo-name`
+
+(fork-repo)=
+## What is forking a GitHub repository?
+
+:::{figure} /images/github/fork-repo.png
+:alt: ""
+
+When you fork a GitHub repository, you make a copy of the files and the commit history into your personal account. This allows you to work on the files independently before suggesting changes through a pull request to make to the parent repository you forked from.
+:::
+
+Using **GitHub.com**, you can create a copy of another user's or organization’s repository—a process called `forking`. When you fork a repository, you get your own version to work on. You can do that without affecting the original files in the parent repository. Others can also fork your repositories, creating their own copies.
+
+Forking is useful because the fork remains linked to the original repository. This allows you to:
+- Update your fork with changes from the original repository.
+- Suggest changes to the original repository, which the owner can review and accept.
+
+By forking, everyone collaborates on their own copies of the project, ensuring the original files stay intact. All changes are tracked in the file history and can be undone if needed. You can fork a repository directly from its main page on **GitHub.com**.
+
+## A few tips about forking a repo
+
+
+1. A GitHub (or GitLab) repository fork is a copy of the GitHub repository that you own in your account. The repository, however, is still connected to the parent repository on GitHub that you forked. This connection means that you can submit pull requests and changes to the repository using your fork.
+
+It also means you can update your fork anytime, as the parent repository is updated to ensure it stays in sync.
+
+2. If you have already forked the repository but some time has passed. You should consider updating or syncing your fork. GitHub has a sync button that you can use to do this (`pyopensci/repo-name`). This will ensure that all of the files in your repository are current.
```

Comment:
re: this, i think it would be worth having a "forks v. branches" subsection in "what is a fork?" section - along the lines of "a fork is an independent copy of a repository, you have a double of all the branches (or maybe just `main` depending on options), in your fork, so if they do work upstream, your version of `main` will get out of date." which would give this a place to live, as well as allow us to discuss stuff like why you should work on branches even in your fork so you can sync `main` when needed and leave links for further reading on rebasing/merging to keep a fork up to date.

Reply:
Yes definitely a good call. I started a graphic that captures this but will add a bit more text either today or next week. We can leave this open for longer if needed to really get the lessons right now that the grant deliverable is in. 
<img width="1277" alt="Screenshot 2025-01-31 at 12 59 36 PM" src="https://github.com/user-attachments/assets/e03f876d-72d0-41c9-8633-81a6dee791ee" />

the last deliverable is community review 🚀 and it makes everything better.